qiskit/transpiler/passes/utils/remove_barriers.py

```diff
@@ -0,0 +1,56 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Remove all barriers in a circuit"""
+
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class RemoveBarriers(TransformationPass):
+    """Return a circuit with any barrier removed.
+
+    This transformation is not semantics preserving.
+
+    Example:
+
+        .. jupyter-execute::
+
+            from qiskit import QuantumCircuit
+            from qiskit.transpiler import PassManager
+            from qiskit.transpiler.passes import RemoveBarriers
+
+            circuit = QuantumCircuit(1)
+            circuit.x(0)
+            circuit.barrier()
+            circuit.h(0)
+
+            pm = PassManager(RemoveBarriers())
+            circuit = pm.run(circuit)
+            circuit.draw()
+
+    """
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        """Run the RemoveBarriers pass on `dag`."""
+
+        # add the merged barriers to a new DAG
+        new_dag = dag._copy_circuit_metadata()
+
+        # go over current nodes, and add them to the new dag
+        for node in dag.topological_op_nodes():
+            if node.name == "barrier":
+                pass
+            else:
+                # copy the condition over too
+                new_dag.apply_operation_back(node.op, qargs=node.qargs, cargs=node.cargs)
```

Comment:
Why not `dag.remove_all_ops_named("barrier")`?

Reply:
@1ucian0 I as not aware of this method, but that is indeed better. I updated the code. 

One question: the method `dag.remove_all_ops_named` removes the operations in-place. Is that OK? I see that the `run()` method for many passes returns a new dag, but the argument dag is modified inplace as well. In the base class for passes, the run method is not annotated with a return value:

https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/basepasses.py#L70

Reply:
I dont think we have any warranty about input preservation. I was trying to play a bit with the pass and it seems like everything seems fine.